File: js/providers/cerebras.js
import { CEREBRAS_API_KEY, SYSTEM_PROMPT } from '../config.js';

export async function fetchCerebras(prompt) {
    const response = await fetch("https://api.cerebras.ai/v1/chat/completions", {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
            "Authorization": `Bearer ${CEREBRAS_API_KEY}`
        },
        body: JSON.stringify({
            model: "llama3.1-8b",
            messages: [
                { role: "system", content: SYSTEM_PROMPT },
                { role: "user", content: prompt }
            ]
            // Cerebras may not support response_format strict json yet, so rely on prompt
        })
    });
    
    if (!response.ok) throw new Error("Cerebras API failed");
    const data = await response.json();
    let content = data.choices[0].message.content;
    try {
        content = content.replace(/```json\n?/g, '').replace(/```\n?/g, '');
        return JSON.parse(content);
    } catch(e) {
        throw new Error("Cerebras returned invalid JSON");
    }
}

File: backend/providers/cerebras.py
"""
Cerebras Provider — Experimental / last-resort backup
Used for: final fallback if all other providers are down
"""

import os
import httpx
from prompt import SYSTEM_PROMPT

CEREBRAS_URL = "https://api.cerebras.ai/v1/chat/completions"
TIMEOUT = 8.0

async def generate_response(message: str, system_prompt: str = SYSTEM_PROMPT) -> str:
    key = os.getenv("CEREBRAS_API_KEY")
    if not key: raise Exception("CEREBRAS_API_KEY missing")
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {key}",
    }
    payload = {
        "model": "llama3.1-8b",
        "messages": [
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": message},
        ],
        "temperature": 0.4,
        "max_tokens": 1024,
    }

    async with httpx.AsyncClient(timeout=TIMEOUT) as client:
        resp = await client.post(CEREBRAS_URL, headers=headers, json=payload)
        resp.raise_for_status()
        data = resp.json()
        return data["choices"][0]["message"]["content"]
